sources/core/Stride.Core/Native/NativeLibraryHelper.cs

```diff
@@ -33,6 +34,24 @@ public static void PreloadLibrary(string libraryName, Type owner)
                     return;
                 }
 
+                // Was the dependency registered beforehand?
+                {
+                    if (NativeDependencies.TryGetValue(libraryName, out var path) && NativeLibrary.TryLoad(path, out var result))
+                    {
+                        LoadedLibraries.Add(libraryName, result);
+                        return;
+                    }
+                }
+
+                // Try default loading mechanism (https://docs.microsoft.com/en-us/dotnet/core/dependency-loading/loading-unmanaged)
+                {
+                    if (NativeLibrary.TryLoad(libraryName, owner.Assembly, searchPath: null, out var result))
```

Comment:
This is the only callsite in this helper that includes owner assembly, do you think we should include it everywhere?

Reply:
It's fine - I thought the `TryLoad` had more overrides and it's not completely clear to me what purpose the passed assembly serves. Like how would it be different if you used `TryLoad(libraryName, out var result)`?

Reply:
Best description I found is in the source code: https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Runtime/InteropServices/NativeLibrary.cs#L121
The assembly defines what AssemblyLoadContext to use. It can also provide search paths through attributes it seems.